### PR TITLE
use goreleaser to publish artifacts

### PR DESCRIPTION
**Pull request by mxyng**
There will soon be a need to publish more artifact. Replace the custom publishing actions with goreleaser which will also handle the other ways of publishing (AUR, Chocolatey)

### CODE REVIEW

File: .github/workflows/cd-binaries.yaml [hunk 1]
```diff
@@ -42,53 +43,12 @@ jobs:
         with:
           distribution: goreleaser
           version: latest
-          args: release --rm-dist --skip-publish --snapshot
+          args: release --rm-dist --snapshot ${{ startsWith(github.ref, 'refs/tags/') || '--skip-publish' }}
```
Comment (mxyng) [inline .github/workflows/cd-binaries.yaml:46]: This allows CI to still build the binaries and packages without publishing them. Useful for running in `main`

**Comment by jmorganca** [comment]
Nice! Good to remove the custom publishing as it's less to maintain.

**Review by jmorganca — APPROVED** [review]

File: .github/workflows/cd-binaries.yaml (hunk 1 above)
Comment (jmorganca) [reply .github/workflows/cd-binaries.yaml:45]: Great. Does this mean we'll have binary artifacts for every commit on `main` (great for testing)

Comment (mxyng) [reply .github/workflows/cd-binaries.yaml:45]: We had this before as well. Those artifacts only live in GitHub actions though

**Pull request merged**